### PR TITLE
dynamic pricing

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
Added logic for dynamic pricing to the fruit board and update/welcome modal. Added marketRate obj based on `Farm.sol`. Divided base prices by current market rate. Let me know if I missed anything and i will update.

### CODE REVIEW

**Review by adamhannigan — APPROVED** [review]
Nice work mate! I tested it out with a range of different hardcoded total supply values and it was updating in the fruit board and total supply boards.

I did notice when we actually plant and harvest the fruits though that the market prices were not in effect. I've put in a commit that ensures we use `marketFruits` in places where we are using the buyPrice and sellPrice :)

It might be worth getting your eyes over that commit as well 👀

File: src/dapp/utils/supply.ts [hunk 1]
```diff
@@ -1,40 +1,70 @@
...
+	if (supply < 500000000) {
+		return 50
+	}
+
+	// Less than 10, 000, 000 tokens
+	if (supply < 1000000000) {
+		return 100
+	}
+
+	// 1 Farm Dollar gets you a 0.00001 of a token - Linear growth from here
```
[the remark this answers is not in the document]
Comment (spencerdezartsmith) [reply src/dapp/utils/supply.ts:68]: Should we just add it in now?

[the remark this answers is not in the document]
Comment (adamhannigan) [reply src/dapp/utils/supply.ts:68]: I'm easy, we never know how fast the supply could grow 🚀

[the remark this answers is not in the document]
Comment (spencerdezartsmith) [reply src/dapp/utils/supply.ts:68]: Adding it now.

[the remark this answers is not in the document]
Comment (spencerdezartsmith) [reply src/dapp/utils/supply.ts:88]: @adamhannigan Can you give this another look over as I had to resolve some conflicts. I just want to make sure all is correct. Tested and the little farm hands are doing their jobs. 💧 🧑‍🌾

**Review by adamhannigan — APPROVED** [review]
LGTM 🤩

**Pull request merged**